transcom/mymove
# MAIN B-20928

[INT PR](https://github.com/transcom/mymove/pull/14058)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20928)

## Summary

Security wanted us to not use `MD5` hashing, so transitioning to `SHA256` for uploads and adding in the `CheckSumAlgorithem` when uploading to S3.

This also contains code to fix some flaky tests. Idk, man.

### How to test

1. You'll need to go to your `.envrc` and uncomment `export STORAGE_BACKEND=s3`
2. Run `direnv allow`
3. Run `aws-vault exec transcom-gov-dev -- make server_run`
4. While that's building, make sure you are in the right AWS profile - `transcom-gov-dev`
6. This allows us to upload files to S3 instead of locally ([docs here](https://dp3.atlassian.net/wiki/spaces/MT/pages/1470955567/How+to+test+storing+data+in+S3+locally))
7. Log in as a customer, fill out all information and upload some orders
8. In your CLI, run `aws-vault login transcom-gov-dev` - this will open the AWS Console
9. Search for S3 and get in there
10. Find your buckets
11. There should be one that says `transcom-gov-dev-app-devlocal-us-gov-west-1`, click it
12. You should see your name/repo if the file uploaded correctly (ex. `daniel.jordan_cn`)
13. Click through and click the file you uploaded to where you see the `Object Overview` section
14. Scroll down and you should see an `Additional checksums` section
15. Confirm you see an additional `SHA-256` checksum

## Screenshots
![Screenshot 2024-10-29 at 11 15 28 AM](https://github.com/user-attachments/assets/9d07671e-e368-4b3f-8062-d6eb486acd8d)
